api/src/main/java/org/infinispan/api/mutiny/MutinyCache.java

```diff
@@ -51,132 +53,366 @@
+   /**
+    * @param map
+    * @param options
+    * @return
+    */
+   Multi<MutinyWriteResult<K>> put(Map<K, V> map, CacheWriteOperationOptions options);
+
+   default Multi<MutinyWriteResult<K>> put(Map<K, V> map) {
+      return put(map, CacheWriteOperationOptions.DEFAULT);
+   }
+
+   /**
+    * Removes a set of keys. Returns the keys that were removed.
+    *
+    * @param keys
+    * @return
+    */
+   default Multi<K> removeAll(Multi<K> keys) {
+      return removeAll(keys, CacheOperationOptions.DEFAULT);
+   }
 
-   Multi<MutinyWriteResult<K>> put(Map<K, V> map);
+   /**
+    * Removes a set of keys. Returns the keys that were removed.
+    *
+    * @param keys
+    * @param options
+    * @return
+    */
+   Multi<K> removeAll(Multi<K> keys, CacheOperationOptions options);
 
    /**
     * Removes a set of keys. Returns the keys that were removed.
     *
     * @param keys
     * @return
     */
-   Multi<K> removeMany(Multi<K> keys);
+   default Multi<CacheEntry<K, V>> getAndRemoveAll(Multi<K> keys) {
+      return getAndRemoveAll(keys, CacheOperationOptions.DEFAULT);
+   }
 
    /**
     * Removes a set of keys. Returns the keys that were removed.
     *
     * @param keys
+    * @param options
     * @return
     */
-   Multi<? extends CacheEntry<K, V>> getAndRemoveMany(Multi<K> keys);
+   Multi<CacheEntry<K, V>> getAndRemoveAll(Multi<K> keys, CacheOperationOptions options);
 
    /**
     * Estimate the size of the store
     *
     * @return Long, estimated size
     */
-   Uni<Long> estimateSize();
+   default Uni<Long> estimateSize() {
+      return estimateSize(CacheOperationOptions.DEFAULT);
+   }
+
+   /**
+    * Estimate the size of the store
+    *
+    * @param options
+    * @return Long, estimated size
+    */
+   Uni<Long> estimateSize(CacheOperationOptions options);
+
+   /**
+    * Clear the store. If a concurrent operation puts data in the store the clear might not properly work
+    *
+    * @return Void
+    */
+   default Uni<Void> clear() {
+      return clear(CacheOperationOptions.DEFAULT);
+   }
 
    /**
     * Clear the store. If a concurrent operation puts data in the store the clear might not properly work
     *
+    * @param options
     * @return Void
     */
-   Uni<Void> clear();
+   Uni<Void> clear(CacheOperationOptions options);
+
+   /**
+    * Executes the query and returns a {@link Multi} with the results
+    *
+    * @param query query String
+    * @return a {@link Multi} which produces query result items.
+    */
+   default <R> Multi<R> find(String query) {
+      return find(query, CacheOperationOptions.DEFAULT);
+   }
 
    /**
     * Executes the query and returns a {@link Multi} with the results
     *
     * @param query query String
     * @return a {@link Multi} which produces query result items.
     */
-   <R> Multi<R> find(String query);
+   <R> Multi<R> find(String query, CacheOperationOptions options);
```

Comment:
missing `findContinuously` when comparing with `AsynCache`... Probably need to decide if we keep `queryContinuously` or `findContinuously`

Reply:
The problem is that `AsyncCache.findContinuously` and `MutinyCache.queryContinuously` both suffer the same issue: they immediately return an event stream, but they don't allow setting query parameters.
We could move the `findContinuously` method to the `Mutiny/Async/Sync/Query` classes. Options like `limit` and `skip` would be ignored for CQs, but I think that it is the cleaner approach.

Reply:
I have moved the `findContinuously` method to the `*Query` interface